docs/proposals/selinux.md

```diff
@@ -0,0 +1,261 @@
+Kubelet managed the SELinux label for `hostPath` volumes, a user who could create a `hostPath`
+volume could affect changes in the state of arbitrary paths within the host's filesystem.  This
+would be a severe security risk, so we will consider hostPath a corner case that the kubelet should
+never perform ownership management for.
+
+#### Shared storage: network
+
+Whether the ownership or SELinux label of a volume for a distributed file system should be managed
+by Kubernetes depends on the cluster operator's policy and the volume type.  Additionally, not all
+distributed file systems support `chown`, `chmod`, and `chcon` operations uniformly.  Our API should
+make it simple to express whether a particular volume should have these concerns managed by
+Kubernetes.  For simplicity's sake, we will assume that if a volume requires ownership management,
+that it also requires relabeling and vice-versa.
+
+#### API requirements
+
+From the above, we know that label management must be applied:
+
+1.  To some volume types always
+2.  To some volume types never
+3.  To some volume types *sometimes*
+
+If the SELinux cluster capability is enabled, volumes should be relabeled with the correct SELinux
+context.  Docker has this capability today; it is desireable for other container runtime
+implementations to provide similar functionality.
+
+Relabeling should be an optional aspect of a volume plugin to accomodate:
+
+1.  volume types for which generalized relabeling support is not sufficient
+2.  testing for each volume plugin individually
+
+## Proposed Design
+
+Our design should minimize code for handling SELinux labelling required in the Kubelet and volume
+plugins.
+
+### Deferral: MCS label allocation
+
+Our short-term goal is to facilitate volume sharing and isolation with SELinux and expose the
+primitives for higher level composition; making these automatic is a longer-term goal.  Allocating
+groups and MCS labels are fairly complex problems in their own right, and so our proposal will not
+encompass either of these topics.  There are several problems that the solution for allocation
+depends on:
+
+1.  Users and groups in Kubernetes
+2.  General auth policy in Kubernetes
+3.  [security policy](https://github.com/GoogleCloudPlatform/kubernetes/pull/7893)
+
+### API changes
+
+We will add a new field to the `PodSecurityContext` to capture the pod-level SELinux context:
+
+```go
+package api
+
+type PodSecurityContext struct {
+	// SELinuxOptions captures the SELinux context for all containers in a Pod
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"
+}
+```
+
+The V1 API will be extended with the same field:
+
+```go
+package v1
+
+type PodSecurityContext struct {
+	// SELinuxOptions captures the SELinux context for all containers in a Pod
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"
+}
+```
+
+#### API backward compatibility
+
+TODO
```

Comment:
@bgrant0607 Also on my list for today, this depended on getting the story right with #12823 before it could be done without wasting time.  The story I'm going to write today is much simpler than the one I would have written when the TODO went in.